### ISSUE TITLE
Issue when using WP preview Tablet and Mobile sizes

### ISSUE THREAD

**Issue by nickfmc** [websevendev/animations-for-blocks#4]
When switching the preview mode to tablet/mobile and back all blocks that have animations enabled will disappear and not come back, without refreshing the page.  This becomes an issue because a lot of gutenberg plugins use these previews as a way to set responsive settings.

**Comment by websevendev** [comment]
I haven't been able to reproduce it.

Can you go to https://playground.wordpress.net/?plugin=animations-for-blocks&url=%2Fwp-admin%2Fpost.php%3Fpost%3D2%26action%3Dedit

It should display "Sample Page" in block editor. Add animation to a paragraph. Switch preview mode to tablet, then back to desktop. For me it works in Firefox 114.0 on Windows.

> all blocks that have animations enabled will disappear and not come back, without refreshing the page

Do they also not re-appear when you switch to Code editor and back?
![image](https://github.com/websevendev/animations-for-blocks/assets/18283192/10809035-7a82-4229-93f2-7d9ff020df88)

**Comment by nickfmc** [comment]
does playground.wordpress.net run natively as if you are viewing it in your browser? or is it powered by a chromium/webkit browser on the backend? I don't know I just ask because it has a windowed feel and webkit like frame around it. it's also not a do this and then this and it will disappear, its very random it seams like the more animations you have on a page the more likely it is to occur more frequently, but I don't know. I tested and yes switching to code editor and back to visual did bring them back. as does just continuing to click back and forth between preview sizes. It may also only occur when you have a slide from left/right something that has to do with x axis overflow. I can put a video together if you want to see it, but there are no console errors when it happens or anything.

**Comment by websevendev** [comment]
> does playground.wordpress.net run natively as if you are viewing it in your browser?

Yes, more info [here](https://developer.wordpress.org/playground/).

> I can put a video together if you want to see it, but there are no console errors when it happens or anything.

That's fine, don't bother, probably not useful.

I'll look into adding a setting that disables the animations in the editor unless you explicitly press Animate block, that should solve the issue.

**Comment by websevendev** [comment]
Added in 1.1.2, you should find an option for it here:
![image](https://github.com/websevendev/animations-for-blocks/assets/18283192/e554b829-04a3-4618-9f0b-1c1da5ce92bd)

Let me know if that works for you.

**Comment by nickfmc** [comment]
Ya, that solve works nice!
